pkgs/by-name/ie/ieda/package.nix

```diff
@@ -0,0 +1,112 @@
+{
+  lib,
+  stdenv,
+  fetchgit,
+  callPackages,
+  cmake,
+  ninja,
+  flex,
+  bison,
+  zlib,
+  tcl,
+  boost,
+  eigen,
+  yaml-cpp,
+  libunwind,
+  glog,
+  gtest,
+  gflags,
+  metis,
+  gmp,
+  python3,
+  onnxruntime,
+}:
+let
+  rootSrc = stdenv.mkDerivation {
+    pname = "iEDA-src";
+    version = "2024-09-10";
+    src = fetchgit {
+      url = "https://gitee.com/oscc-project/iEDA";
+      rev = "a68b691b9d25fafd8c10fae3df7ef3837a42e052";
+      sha256 = "sha256-0rSESfNqI3ALipNAInwcYSccq9C0WuXI9na44TyYAgY=";
+    };
+
+    patches = [
+      ./fix-bump-gcc.patch
+
+      # We need to build rust projects with rustPlatform
+      # and remove hard coded linking to libonnxruntime
+      ./remove-subprojects-from-cmake.patch
+    ];
+
+    dontBuild = true;
+    dontFixup = true;
+    installPhase = ''
+      cp -r . $out
+    '';
+
+  };
+
+  rustpkgs = callPackages ./rustpkgs.nix { inherit rootSrc; };
+in
+stdenv.mkDerivation {
+  pname = "iEDA";
+  version = "0-unstable-2024-10-11";
+
+  src = rootSrc;
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    flex
+    bison
+    zlib
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "CMD_BUILD" true)
```

Comment:
```suggestion
```
CMake claims it is not used by the build. Though maybe I'm misreading the log.

Reply:
I think it's at least used [here](https://gitee.com/oscc-project/iEDA/blob/master/src/apps/CMakeLists.txt#L4) and [here](https://gitee.com/oscc-project/iEDA/blob/master/CMakeLists.txt#L54), so we need this to avoid CMAKE_RUNTIME_OUTPUT_DIRECTORY been overridden (this project uses CMAKE_RUNTIME_OUTPUT_DIRECTORY as destination of executables). Not sure about why CMake warns this.